cluster/cluster.go

```diff
@@ -40,6 +42,53 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 	}, nil
 }
 
+type EtcdConfig struct {
```

Comment:
this will be moved out once i figure out part2

Reply:
I don't quite understand why we can't use the `embed.Config` for this. https://godoc.org/github.com/etcd-io/etcd/embed#Config